src/BloomExe/Book/Book.cs

```diff
@@ -143,25 +144,48 @@ public string TitleBestForUserDisplay
 		{
 			get
 			{
-				var list = new List<string>();
-				list.Add(_collectionSettings.Language1Iso639Code);
-				if (_collectionSettings.Language2Iso639Code != null)
-					list.Add(_collectionSettings.Language2Iso639Code);
-				if (_collectionSettings.Language3Iso639Code != null)
-					list.Add(_collectionSettings.Language3Iso639Code);
-				list.Add("en");
-
 				var title = _bookData.GetMultiTextVariableOrEmpty("bookTitle");
-				var t = title.GetBestAlternativeString(list);
-				if(string.IsNullOrEmpty(t))
+				var display = title.GetExactAlternative(_collectionSettings.Language1Iso639Code);
+				
+				if (string.IsNullOrEmpty(display))
 				{
-					return "Title Missing";
+					//if this book is one of the ones we're editing in our collection, it really
+					//needs a title in our main language, it would be confusing to show a title from some other langauge
+					if (IsEditable ||  title.Empty)
+					{
+						display = LocalizationManager.GetString("CollectionTab.TitleMissing", "Title Missing",
+							"Shown as the thumbnail caption when the book doesn't have a title");
+					}
+					//but if this book is just in our list of sources, well then let's look through the names
+					//and try to get one that is likely to be helpful
+					else
+					{
+						var orderedPreferences = new List<string>();
+						orderedPreferences.Add(LocalizationManager.UILanguageId);
+
+						orderedPreferences.Add(_collectionSettings.Language1Iso639Code);
+						if (_collectionSettings.Language2Iso639Code != null)
+							orderedPreferences.Add(_collectionSettings.Language2Iso639Code);
+						if (_collectionSettings.Language3Iso639Code != null)
+							orderedPreferences.Add(_collectionSettings.Language3Iso639Code);
+
+						orderedPreferences.Add("en");
+						orderedPreferences.Add("fr");
+						orderedPreferences.Add("es");
```

Comment:
Is it worth adding a few more major languages? In case people start authoring in Indonesia, Korea, China, or Russia, to pick a few that seem vaguely likely.